clean/ca/los_angeles_pd.py

```diff
@@ -0,0 +1,205 @@
+                                "asset_url": link["href"].replace("\n", ""),
+                                "name": link.text.strip().replace("\n", ""),
+                            }
+                            metadata.append(payload)
+        # Store the metadata in a JSON file in the data directory
+        outfile = self.data_dir.joinpath(f"{self.agency_slug}.json")
+        self.cache.write_json(outfile, metadata)
+        # Return path to metadata file for downstream use
+        return outfile
+
+    def _get_child_page(self, index_page: Path, throttle: int = 0) -> List[dict]:
+        """Get URLs for child pages from index pages."""
+        html = self.cache.read(index_page)
+        soup = BeautifulSoup(html, "html.parser")
+        # Get all the child page URLs
+        parent_div = soup.find("div", class_="view-content")
+        links = parent_div.find_all("a")  # type: ignore
+        child_pages = []
+        for anchor in links:
+            time.sleep(throttle)
+            page_meta = {
+                "source_index_page": index_page,  # index page where this child page was found
+                "source_name": anchor.text.strip(),
+                "url": urllib.parse.urljoin(self.base_url, anchor.attrs["href"]),
+            }
+            page_meta["cache_name"] = (
+                f"{page_meta['source_name'].replace(' ', '_')}.html"
+            )
+            page_meta.update(
+                urllib.parse.parse_qs(urllib.parse.urlparse(page_meta["url"]).query)
+            )
+            # Stash child pages in folder matching name of index page where it's listed
+            # Construct index page directory
+            index_page_dir = f"{self.agency_slug}/{index_page.stem}"
+            # Construct local file path inside index page directory
+            relative_path = f"{index_page_dir}/{page_meta['cache_name']}"
+            # Download the child page
+            cache_path = self.cache.download(relative_path, page_meta["url"], "utf-8")
+            # Update page metadata with full path in cache and relative path
+            page_meta.update(
+                {
+                    "cache_path": cache_path,
+                    "relative_path": relative_path,
+                }
+            )
+            child_pages.append(page_meta)
+        return child_pages
+
+    def _get_index_page_urls(self, first_index_page: Path) -> List[str]:
+        """Get the URLs for all index pages."""
+        # Read the cached HTML file for home page
+        html = self.cache.read(first_index_page)
+        soup = BeautifulSoup(html, "html.parser")
+        # Gross, but necessary to pass mypy type checking
+        complaint_table = (
+            soup.find("section", class_="cmn-column cmn-three-column-grid cmn-widget-grid  wow fadeInUp")  # type: ignore
+            .a.attrs["href"]  # type: ignore
+            #.split("https://www.lapdonline.org")[-1]  # type: ignore
```

Comment:
Let's talk about deleting these lines once we have a handle on this section's implementation.